frontend/src/visualization/filtering/Filter.tsx

```diff
@@ -85,64 +77,30 @@ const Filter = (): JSX.Element => {
   const [tabIndex, setTabIndex] = React.useState(0);
   const [open, setOpen] = React.useState(false);
 
-  const filterStateStore = useService(FilterStateStore);
+  const schemaStore = useService(SchemaStore);
 
   const entityStyleStore = useService(EntityStyleStore);
   const styleProvider = useObservable(
     entityStyleStore.getState(),
     entityStyleStore.getValue()
   );
 
-  const schemaService = useService(SchemaService);
-
-  const loadingStore = useService(LoadingStore);
-  const errorStore = useService(ErrorStore);
-
-  const schema = useObservable(
-    forkJoin([
-      from(schemaService.getNodeTypes()),
-      from(schemaService.getEdgeTypes()),
-    ]).pipe(
-      withLoadingBar({ loadingStore }),
-      withErrorHandler({ errorStore }),
-      map((schemaFromService) => ({
-        nodes: schemaFromService[0],
-        edges: schemaFromService[1],
-      }))
-    ),
-    { nodes: [], edges: [] }
-  );
+  const schema = useObservable(schemaStore.getState(), {
+    nodeTypes: [],
+    edgeTypes: [],
+  });
 
-  // filterStore will only be filled with initial FilterLineStates on the first render.
   useEffect(() => {
-    const nodeLineStates: FilterLineState[] = [];
-    const edgeLineStates: FilterLineState[] = [];
-
-    for (const nodeTypes of schema.nodes) {
-      nodeLineStates.push({
-        type: nodeTypes.name,
-        isActive: false,
-        propertyFilters: [],
-      });
-    }
-
-    for (const edgeTypes of schema.edges) {
-      edgeLineStates.push({
-        type: edgeTypes.name,
-        isActive: false,
-        propertyFilters: [],
-      });
+    // only update the schemaStore on the very first time, the Filter is rendered.
+    // So no update happens when switching between tabs.
+    if (!schemaStore.initialized) {
```

Comment:
Such information should not be published. Thus the SchemaStore should be changed a bit (use ensureInit functionality); See change requests